test/network/key-rotation.js

```diff
@@ -0,0 +1,175 @@
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+const {
+  Hbar,
+  Status,
+  PrivateKey,
+  AccountInfoQuery,
+  AccountCreateTransaction,
+  AccountUpdateTransaction,
+} = require('@hashgraph/sdk');
+const { expect } = require('chai');
+const Utils = require('../system-contracts/hedera-token-service/utils');
+
+describe('Key Rotation Test Suite', function () {
+  let client;
+  let accountId_Alpha;
+  let accountPrivateKey_Alpha;
+  let accountPublicKey_Alpha;
+  let accountEvmAddress_Alpha;
+
+  beforeEach(async function () {
+    client = await Utils.createSDKClient();
+
+    // Generate a new ECDSA keys
+    accountPrivateKey_Alpha = PrivateKey.generateECDSA();
+    accountPublicKey_Alpha = accountPrivateKey_Alpha.publicKey;
+    accountEvmAddress_Alpha = accountPublicKey_Alpha.toEvmAddress();
+
+    // Create new account and assign the ECDSA public key as admin key
+    const newAccountAlphaTx = await new AccountCreateTransaction()
+      .setKey(accountPublicKey_Alpha)
+      .setInitialBalance(Hbar.fromTinybars(1000))
+      .setAlias(accountEvmAddress_Alpha)
+      .execute(client);
+
+    // Get the new account ID
+    const newAccountAlphaTxRceipt = await newAccountAlphaTx.getReceipt(client);
+    accountId_Alpha = newAccountAlphaTxRceipt.accountId;
+  });
+
+  it('Should remain the same EVM key alias after key rotation with a different ECDSA key', async function () {
+    // Generate a new ECDSA key
+    const accountPrivateKey_Beta = PrivateKey.generateECDSA();
+    const accountPublicKey_Beta = accountPrivateKey_Beta.publicKey;
+    const accountEvmAddress_Beta = accountPublicKey_Beta.toEvmAddress();
+
+    // Create the transaction to rotate and change the key to a different ECDSA key
+    const accountUpdateTransaction = new AccountUpdateTransaction()
+      .setAccountId(accountId_Alpha)
+      .setKey(accountPublicKey_Beta)
+      .freezeWith(client);
+
+    // Sign with the old key
+    const signTxByAlpha = await accountUpdateTransaction.sign(
+      accountPrivateKey_Alpha
+    );
+
+    // Sign with the new key
+    const signTxByBeta = await signTxByAlpha.sign(accountPrivateKey_Beta);
+
+    // Submit the transaction
+    const txResponse = await signTxByBeta.execute(client);
+
+    // Get the receipt of the transaction
+    const receipt = await txResponse.getReceipt(client);
+
+    // Get the transaction consensus status
+    const transactionStatus = receipt.status;
+    expect(transactionStatus).to.equal(Status.Success);
+
+    // Check the account info after key rotation
+    const accountInfoAfterKeyRotation = await new AccountInfoQuery()
```

Comment:
Please print to console the accountId, publickey and evm address after

Reply:
Updated. Also added log output in PR description